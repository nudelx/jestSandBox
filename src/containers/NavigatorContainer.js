import React from 'react'
import { connect } from 'react-redux'
import Nav from '../componets/navigator.js'
import NavButtonActionsCreator from '../redux/action/navButtonAction.js'
import { fetchingMoviewProcess } from '../redux/action/movieActionCreator.js'

const NavigatorContainer = (props) => {
  const onCLick = (page) => {
    console.log('click')
    props.createNavButtonClick(page)
    props.fetchingMoviewProcess(page)
  }
  return ( <Nav onClick={onCLick} /> )
}

const mapDispatchToProps = {
  fetchingMoviewProcess,
  createNavButtonClick: NavButtonActionsCreator.createNavButtonClick,
}

export default connect( null, mapDispatchToProps)(NavigatorContainer)
