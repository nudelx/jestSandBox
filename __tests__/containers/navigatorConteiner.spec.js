import React from  'react'
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme'
import NavigatorContainer from '../../src/containers/navigatorContainer'
import toJson from 'enzyme-to-json';
import enzymeSerializer from 'enzyme-to-json/serializer';
import configureStore from 'redux-mock-store'
import Nav from '../../src/componets/navigator'

const initialState = { content: 'test_1', movies: {movie:'test'}}
const mockStore = configureStore()
let store = mockStore(initialState)
let container = null


describe('NavigatorContainer test ', () => {

  beforeEach(()=>{
    store = mockStore(initialState)
    store.dispatch = jest.fn()
    container = mount(<NavigatorContainer store={store} /> )
 })

  it('should render without crashing with Nav', () => {
    expect(toJson(container)).toMatchSnapshot()
  })

  it('shold render navigator', () => {
    expect(toJson(container)).toMatchSnapshot()
    const nav = container.find(Nav)
    nav.find('li').map(item =>  item.simulate('click'))
    expect(store.dispatch.mock.calls).toMatchSnapshot()
  })

  it('shold render dispatch double action ', () => {
    const nav = container.find(Nav)
    nav.find('li').map(item =>  item.simulate('click'))
    expect(store.dispatch.mock.calls).toMatchSnapshot()
  })
})
