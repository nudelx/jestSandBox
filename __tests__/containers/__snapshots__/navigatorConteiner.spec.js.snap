// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigatorContainer test  shold render dispatch double action  1`] = `
Array [
  Array [
    Object {
      "page": "page_1",
      "type": "NAV_BUTTON::CLICK",
    },
  ],
  Array [
    [Function],
  ],
  Array [
    Object {
      "page": "page_2",
      "type": "NAV_BUTTON::CLICK",
    },
  ],
  Array [
    [Function],
  ],
  Array [
    Object {
      "page": "page_3",
      "type": "NAV_BUTTON::CLICK",
    },
  ],
  Array [
    [Function],
  ],
]
`;

exports[`NavigatorContainer test  shold render navigator 1`] = `
<Connect(NavigatorContainer)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <NavigatorContainer
    createNavButtonClick={[Function]}
    fetchingMoviewProcess={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Nav
      onClick={[Function]}
    >
      <div
        className="nav-bar"
      >
        <ul>
          <li
            onClick={[Function]}
          >
            Page 1
          </li>
          <li
            onClick={[Function]}
          >
            Page 2
          </li>
          <li
            onClick={[Function]}
          >
            Page 3
          </li>
        </ul>
      </div>
    </Nav>
  </NavigatorContainer>
</Connect(NavigatorContainer)>
`;

exports[`NavigatorContainer test  shold render navigator 2`] = `
Array [
  Array [
    Object {
      "page": "page_1",
      "type": "NAV_BUTTON::CLICK",
    },
  ],
  Array [
    [Function],
  ],
  Array [
    Object {
      "page": "page_2",
      "type": "NAV_BUTTON::CLICK",
    },
  ],
  Array [
    [Function],
  ],
  Array [
    Object {
      "page": "page_3",
      "type": "NAV_BUTTON::CLICK",
    },
  ],
  Array [
    [Function],
  ],
]
`;

exports[`NavigatorContainer test  should render without crashing with Nav 1`] = `
<Connect(NavigatorContainer)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <NavigatorContainer
    createNavButtonClick={[Function]}
    fetchingMoviewProcess={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Nav
      onClick={[Function]}
    >
      <div
        className="nav-bar"
      >
        <ul>
          <li
            onClick={[Function]}
          >
            Page 1
          </li>
          <li
            onClick={[Function]}
          >
            Page 2
          </li>
          <li
            onClick={[Function]}
          >
            Page 3
          </li>
        </ul>
      </div>
    </Nav>
  </NavigatorContainer>
</Connect(NavigatorContainer)>
`;
